src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -1052,35 +1052,13 @@ void SerialHeap::print_heap_change(const PreGenGCValues& pre_gc_values) const {
   MetaspaceUtils::print_metaspace_change(pre_gc_values.metaspace_sizes());
 }
 
-class GenGCPrologueClosure: public SerialHeap::GenClosure {
- private:
-  bool _full;
- public:
-  void do_generation(Generation* gen) {
-    gen->gc_prologue(_full);
-  }
-  GenGCPrologueClosure(bool full) : _full(full) {};
-};
-
 void SerialHeap::gc_prologue(bool full) {
   assert(InlineCacheBuffer::is_empty(), "should have cleaned up ICBuffer");
 
   // Fill TLAB's and such
   ensure_parsability(true);   // retire TLABs
 
-  // Walk generations
-  GenGCPrologueClosure blk(full);
-  generation_iterate(&blk, false);  // not old-to-young.
-};
-
-class GenGCEpilogueClosure: public SerialHeap::GenClosure {
- private:
-  bool _full;
- public:
-  void do_generation(Generation* gen) {
-    gen->gc_epilogue(_full);
-  }
-  GenGCEpilogueClosure(bool full) : _full(full) {};
+  _old_gen->gc_prologue();
```

Comment:
Why is this not calling `_young_gen->gc_prolog(full)`?

Reply:
Young-gen doesn't have such method. (It was using the empty one from super class.)

Reply:
Got it. Thanks.